src/libtriton/arch/arm/arm32/arm32Cpu.cpp

```diff
@@ -241,45 +244,81 @@ namespace triton {
               inst.setUpdateFlag(detail->arm.update_flags);
 
               /* FIXME: Quick (and super ugly) hack. Capstone is reporting
-               * update_flags equals true for ADC instruction when it shouldn't
-               * (it should only report true for ADCS).
+               * update_flags equals true for ADC, RSC and SBC instruction when
+               * it shouldn't (it should only report true when the S suffix is
+               * present).
                */
-              if (inst.getDisassembly().find("adc") == 0 && inst.getDisassembly().at(3) != 's') {
+              if ((inst.getDisassembly().find("adc") == 0 && inst.getDisassembly().at(3) != 's') ||
+                  (inst.getDisassembly().find("rsc") == 0 && inst.getDisassembly().at(3) != 's') ||
+                  (inst.getDisassembly().find("sbc") == 0 && inst.getDisassembly().at(3) != 's')) {
                 inst.setUpdateFlag(false);
               }
```

Comment:
For the future we really have to fix this, because it implies an big overhead...

Reply:
Indeed. I leave here the Capstone's issue for future reference: https://github.com/aquynh/capstone/issues/1568

In the meantime, I'll refactor this code into a `postDisassembly` function.

Reply:
Refactored in commit dbe558c2312fb3b5c875bee8d1cad1c0e9af5a70.